### ISSUE TITLE
Form (custom, or drag and drop) does not keep the table atribute order.

### ISSUE THREAD

**Issue by kylefelipe** [qgis/QGIS#30088]
**Describe the bug**
If we set a order on a attribute table, when change to form mode, the preview panel (left) changes the order to the chosen field or expression.

Example: 
Table mode order by "id"
![image](https://user-images.githubusercontent.com/24795382/58929622-507b1600-872e-11e9-8b2b-1a96e2cc7ca4.png)

Form with expression preview "id" || ' - ' || "client"

![image](https://user-images.githubusercontent.com/24795382/58929706-a18b0a00-872e-11e9-99f2-d89aa2910f5a.png)

left panel does not keep the table mode order.
The same happens with relations.

**QGIS and OS versions**

QGIS Version: 3.4
OS Version: Windows 7 an Debian Stretch

With Qgis 2.18 the form keeps the table order, same for relations.

Sample project with data:
[sample_data.zip](https://github.com/qgis/QGIS/files/3255478/sample_data.zip)

**Comment by andreasneumann** [comment]
I don't thint this is a bug. The sort order between table view and form view is independent - on purpose.

Did you use the "sort by preview expression" in the form mode?

**Comment by DelazJ** [comment]
Furthermore, your expression returns a string while id is an integer so you will not get the same order in this case afaik (`9<10` but `'9' >'10'`).

**Comment by andreasneumann** [comment]
> Furthermore, your expression returns a string while id is an integer so you will not get the same order in this case afaik (`9<10` but `'9' >'10'`).

Yes - you could introduce leading zeros to get proper sorting in those strings.

**Comment by kylefelipe** [comment]
@Andrea
> Did you use the "sort by preview expression" in the form mode?
R. No.

Well, some cases its a nice idea keep the same order. 
Here in company this is very handfull, because its helps our staff to find issues during the registration.
They can sort ascending or descending, no mather what field they choose to sort.
I'm trying to migrate our project here to QGIS3.4 from 2.18, version that they can do this. We use to multi-purpose registry.

**Comment by 3nids** [comment]
Hitting something similar this week, I believe the sorting in the form mode should/could offer: reverse sorting, sorting by expression and/or potentially follow order of attribute table
